src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -48,6 +48,11 @@ internal AssemblyScanner(Assembly assemblyToScan)
         /// </summary>
         public bool ScanAppDomainAssemblies { get; set; } = true;
 
+        /// <summary>
+        /// Determines if the scanner should scan assemblies from the application directory.
+        /// </summary>
+        public bool ScanFileSystemAssemblies { get; set; } = true;
+
         internal string CoreAssemblyName { get; set; } = NServicebusCoreAssemblyName;
```

Comment:
@timbussmann I don't get this... `AssemblyScanner` is already in Core, why is this string used? Shouldn't typeof(AssemblyScanner).GetType().Assembly.GetName().Name`?

```suggestion
        // Exists for testing backwards compatibility
        internal string CoreAssemblyName { get; set; } = typeof(AssemblyScanner).GetType().Assembly.GetName().Name;
```

Or why isn't this even `Asssembly` instead of `string?` Would make the comparison later on also be comparing `Assembly` instead of `string`.


Property  `CoreAssemblyName`  is only used to override the name... but I don't get the tests and why the `DynamicAssembly` is needed. Are these actually useful?

Reply:
This is required to make the testing flexible enough to test scenarios that for example simulate different versions of core. It's specifically coded to enable testing so it's a bit ugly but allows much more complex testing scenarios.